src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,142 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList implements Iterable<Task> {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Initializes an empty task list.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Initializes the task list with the given tasks.
+     *
+     * @param tasks The initial list of tasks.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task The task to be added.
+     */
+    public void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Retrieves a task from the list.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return The task at the specified index.
+     */
+    public Task get(int index) {
+        return tasks.get(index);
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param index The index of the task to be deleted.
+     */
+    public void delete(int index) {
```

Comment:
Nit: perhaps return the Task? seems like behaviour that makes sense, even though I do think IntelliJ will draw a squiggly.